akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSplitWhenSpec.scala

```diff
@@ -257,10 +257,11 @@ class FlowSplitWhenSpec extends StreamSpec {
     }
 
     "fail substream if materialized twice" in assertAllStagesStopped {
+      import system.dispatcher
       an[IllegalStateException] mustBe thrownBy {
         Await.result(
           Source.single(1).splitWhen(_ ⇒ true).lift
-            .mapAsync(1) { src ⇒ src.runWith(Sink.ignore); src.runWith(Sink.ignore) } // Sink.ignore+mapAsync pipes error back
+            .mapAsync(1) { src ⇒ src.runWith(Sink.ignore).flatMap(_ ⇒ src.runWith(Sink.ignore)) } // Sink.ignore+mapAsync pipes error back
```

Comment:
why would the order matter? shouldn't it throw also when materialized twice at the same time?